### DISCUSSION TITLE
Stock Plugin always return 403 (Not Subscribed to the API)

### DISCUSSION THREAD

**Discussion by billpwchan** [lowlighter/metrics#931] [Q&A] [answered]
Hi! 😃

Thanks for this amazing repo to make my README really stands out from others lol. 

However, when trying to use the stock plugin, it always returns 
![image](https://user-images.githubusercontent.com/25129943/156918534-4a05ac10-c719-4690-92d0-5e7400306c4f.png)

My configuration for this plugin is:
![image](https://user-images.githubusercontent.com/25129943/156918569-bdb7eb5e-0cdc-4285-bef6-83d722de4a2d.png)

Meanwhile, I'm sure that I've subscribed to the required service in Rapid API, as can seen in the following screenshot. 
![image](https://user-images.githubusercontent.com/25129943/156918619-800841c4-47ca-4548-99a8-96283337f8c3.png)

Would love to have your help to check if there is anything wrong with my config, or is it related to another issue. Thanks!

**Comment by lowlighter** [comment, marked as answer]
Hi 👋 !

It looks fine, I don't remember doing additional stuff to configure this plugin 😕 
Could you retry putting the content of `x-rapidapi-key` in your `STOCK_TOKEN` repository secrets ? (ensure there's no trailing spaces)

From what I tested locally, it seems that the rapid api also returns this error if the token is invalid so it's not possible to differentiate between a token error or a missing subscription

Thanks for your support !

**Reply by billpwchan** [reply]
Thanks for your answer! Yet I've tried it again with confidence that there is no trailing space included in the key, and I still get this 403 error. 

When I used the Test Endpoint feature in RapidAPI, I can get the response successfully so that shouldn't be an issue from the rapidapi side. 

![image](https://user-images.githubusercontent.com/25129943/157580180-aa408338-7bc7-4591-befc-19e7eeb4aa2c.png)

Not sure where is the cause of this issue...May I ask if you can successfully try this stock plugin using GitHub actions? Or you would recommend me to use another approach to render this stock svg? 😶‍🌫️

**Reply by lowlighter** [reply]
Could you check that you also have a pricing plan selected (even if it's the free one) ?

According to this discussion, it seems possible to subscribe to the API without actually selecting a plan: (https://rapidapi.com/Top-Rated/api/e-mail-check-invalid-or-disposable-domain/discussions/9487) which will result in the same error you got.

Can you check the pricing tab: 
![image](https://user-images.githubusercontent.com/22963968/157772317-877c099a-55a6-4ae7-b3ad-1fc18a066a36.png)

I think it is supposed to also appear in your account overview:
![image](https://user-images.githubusercontent.com/22963968/157772855-4bbd8f5e-f529-42a2-ad75-796ca8c1707e.png)

**Reply by billpwchan** [reply]
Thanks a lot for the help! Yet I'm still not able to load the stock prices even after registering a new Rapid API account and subscribing to this plan. Let me investigate this further to see if this is an issue with the API or with the configuration. 😊

**Reply by billpwchan** [reply]
@lowlighter Finally found the problem, but might need some help from your side as well (if you're available lol~)

Seems like to reason why it always show 403 error to me is because the X-RapidAPI-Host is different in the latest Yahoo Finance API. From the README in your repository, the X-RapidAPI-Host is **apidojo-yahoo-finance...,** whereas in the latest one the value has been changed to **yh-finance**...Let me know whether this issue can be fixed by updating the host in the code? Thanks!

![image](https://user-images.githubusercontent.com/25129943/168945500-a92d2ef6-6c6a-402c-add7-b183d7f98148.png)
![image](https://user-images.githubusercontent.com/25129943/168945606-3046ffba-7a56-4991-9ef9-fef28bb41a0f.png)
